doc/whatsnew/3/3.2/index.rst

```diff
@@ -12,5 +12,77 @@
 Summary -- Release highlights
 =============================
 
-
 .. towncrier release notes start
+
+What's new in Pylint 3.2.0?
+---------------------------
+Release date: 2024-05-14
+
+
+New Features
+------------
+
+- Understand `six.PY2` and `six.PY3` for conditional imports.
+
+  Closes #3501 (`#3501 <https://github.com/pylint-dev/pylint/issues/3501>`_)
+
+- A new `github` reporter has been added. This reporter  returns the output of `pylint` in a format that
+  Github can use to automatically annotate code. Use it with `pylint --output-format=github` on your Github Workflows.
+
+  Closes #9443. (`#9443 <https://github.com/pylint-dev/pylint/issues/9443>`_)
+
+
+
+New Checks
+----------
+
+- Add check ``possibly-used-before-assignment`` when relying on names after an ``if/else``
+  switch when one branch failed to define the name, raise, or return.
+
+  Closes #1727 (`#1727 <https://github.com/pylint-dev/pylint/issues/1727>`_)
+
+- Checks for generators that use contextmanagers that don't handle cleanup properly.
+  Is meant to raise visibility on the case that a generator is not fully exhausted and the contextmanager is not cleaned up properly.
+  A contextmanager must yield a non-constant value and not handle cleanup for GeneratorExit.
+  The using generator must attempt to use the yielded context value `with x() as y` and not just `with x()`.
+
+  Closes #2832 (`#2832 <https://github.com/pylint-dev/pylint/issues/2832>`_)
+
+
+
+False Negatives Fixed
+---------------------
+
+- If and Try nodes are now checked for useless return statements as well.
+
+  Closes #9449. (`#9449 <https://github.com/pylint-dev/pylint/issues/9449>`_)
+
+- Fix false negative for ``property-with-parameters`` in the case of parameters which are ``positional-only``, ``keyword-only``, ``variadic positional`` or ``variadic keyword``.
+
+  Closes #9584 (`#9584 <https://github.com/pylint-dev/pylint/issues/9584>`_)
+
+
+Internal Changes
+----------------
+
+- Update astroid version to 3.2.0.
```

Comment:
I always wonder about shouting about the astroid changes. One major improvement people were waiting for was the fix for `@overload`. Do you want to mention it here someplace?

Reply:
Yeah translating that into the implication for pylint would be better for sure.

Reply:
How about that (in false positives fixed):

```
pylint now understand the ``@overload`` decorator return values better.

Closes #4696
```

Reply:
I dig it. I'd go with "understands" 👍